### ISSUE TITLE
no attribute 'betas8'

### ISSUE THREAD

**Issue by YOKE** [NoamRosenberg/AutoML#61]
When i try to run CUDA_VISIBLE_DEVICES=0 python decode_autodeeplab.py --dataset cityscapes --resume /AutoDeeplabpath/checkpoint.pth.tar
There is an error:
![image](https://user-images.githubusercontent.com/34132063/66564592-ecaca200-eb92-11e9-8776-8ebb2fa13eb7.png)
I can't find betas8, betas16....
Could you help me? Thanks a lot~ @NoamRosenberg @zhizhangxian

**Comment by zhizhangxian** [comment]
Now, you can use branch temp for search and retrain now
But there are still some small bugs in the viterbi_decode, you must multiply beta with 1e3 before doing viterbi. We will fix it in next version
Thanks

**Comment by YOKE** [comment]
> Now, you can use branch temp for search and retrain now
> But there are still some small bugs in the viterbi_decode, you must multiply beta with 1e3 before doing viterbi. We will fix it in next version
> Thanks

I compared some files of .py  in branch temp and branch master, and found no differences.
The problem of betas8 still confuses me. I can't find the attribute betas8 of model.
Could you help me? Thanks!

**Comment by zhizhangxian** [comment]
Yes, I checked the files in Temp branch.
For some reasons, the files in this branch is not updated, sorry.
beta8 and beta16, etc is the arch_parameters in previous version and now they wont be used and are all replaced by betas
Looks like the betas in test branch can work. Maybe you can use the 'test' branch for now. And I will fix the Temp branch today. Thanks

**Comment by zhizhangxian** [comment]
ohh, I checked your issue again, seems I misunderstand your meaning. 
We now only fix the viterbi in decode_formula.py but not the decode_autodeeplab.py
actually the viterbi inputs the betas matrix(torch.tensor(12,4,3)) and output the network_arch(np.array(12)), so for now we directly use the class Decoder in decode_formula.py and send the alphas and betas by hand. Thanks for your advice, we will fix the decode_autodeeplab soon, Maybe today, thanks!

**Comment by YOKE** [comment]
Thanks for your reply~
For sending alphas and betas by hand, bottom_betas=betas[:,0,:], betas8=betas[:,1,:], betas16=betas[:,2,:], top_betas=betas[:,3,:]?  Am i right?
If doing like this, channel mismatch would occur.
![image](https://user-images.githubusercontent.com/34132063/66622061-e4974580-ec18-11e9-8ffc-aac894cc367f.png)
network_space[1:, 0, 1:].shape=[11, 2], but normalized_bottom_betas.shape=[12, 3]
Could you tell me the details about this problem? Thanks a lot~

**Comment by zhizhangxian** [comment]
No, the new betas can not compatible beta_4,8 bottom and top. because betas is not the concat of them. Actually the betas means the cofficients of information out from each cell and old betas means the information input each cell. So if you want to decode a network, you should use new autodeeplab.py to search and get alphas and betas.

**Comment by YOKE** [comment]
> No, the new betas can not compatible beta_4,8 bottom and top. because betas is not the concat of them. Actually the betas means the cofficients of information out from each cell and old betas means the information input each cell. So if you want to decode a network, you should use new autodeeplab.py to search and get alphas and betas.
Oh, how can i get the new autodeeplab.py?

**Comment by zhizhangxian** [comment]
There are three branchs now:
1. Master: the betas for viterbi decode in this branch is wrong, but in autodeeplab.py is right
2.Temp: it is ahead of Master for viterbi decode beta, but the decode_autodeeplab.py is not right now, but know we wont update this branch so it is stable now
3. test: I will commit fixes on this branch, but is is the most unstable branch
If you wanna get new autodeeplab.py, i recommend to use Temp branch
https://github.com/NoamRosenberg/AutoML/blob/Temp/auto_deeplab.py
